### PR TITLE
Modernize the AppStream metadata

### PR DESCRIPTION
**Pull request by AsciiWolf**
Done as a part of my MilkyTracker Flatpak work (see #334), but will also be useful here in upstream. The currently shipped AppData file is outdated, contains many deprecated tags and is even missing some basic tags. This PR fixes it (and also renames the .appdata file to a proper name using rDNS ID and correct `.metainfo.xml` suffix.

### CODE REVIEW

**Comment by AsciiWolf** [comment]
```
$ appstreamcli validate resources/org.milkytracker.MilkyTracker.metainfo.xml

✔ Validation was successful: pedantic: 3
```

File: appveyor.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ for:
   - sh: cp ../resources/pictures/carton.png milkytracker.png
   - sh: cp ../resources/milkytracker.desktop .
   - sh: sed -i 's|MilkyTracker|Milkytracker|g;s|Comment=.*|Comment=A Portable SampleTracker DAW|g' milkytracker.desktop
-  #- sh: cp ../resources/milkytracker.appdata /usr/share/metainfo
+  #- sh: cp ../resources/org.milkytracker.MilkyTracker.metainfo.xml /usr/share/metainfo
```
Comment (hfiguiere) [inline appveyor.yml:62]: they should be installed when running install. (ie if they are not then it should be fixed to do so)

Comment (AsciiWolf) [reply appveyor.yml]: I am not the MilkyTracker contributor and don't want to modify their own CI/AppImage build manifests. Just fixing the commented out line should be enough for now, if MilkyTracker developers want, they can fix the rest themselves.

**Comment by AsciiWolf** [comment]
/cc @Deltafire

Any chance you could take a look at this? It is already used for the Flatpak build of MilkyTracker and there are no problems. It should be fine to be merged. ;-)

**Pull request merged**